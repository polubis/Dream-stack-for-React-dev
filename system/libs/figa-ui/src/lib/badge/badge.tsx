import c from 'classnames';
import type { BadgeProps } from './defs';

const Badge = ({
  className,
  children,
  variant = 'filled',
  motive = 'primary',
  ...props
}: BadgeProps) => {
  return (
    <div className={c('badge', className, variant, motive)} {...props}>
      {children}
    </div>
  );
};

export { Badge };
