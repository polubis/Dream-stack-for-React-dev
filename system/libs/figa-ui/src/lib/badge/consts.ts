const BADGE_VARIANTS = ['filled', 'outlined'] as const;
const BADGE_MOTIVES = ['primary', 'secondary', 'ok'] as const;

export { BADGE_VARIANTS, BADGE_MOTIVES };
