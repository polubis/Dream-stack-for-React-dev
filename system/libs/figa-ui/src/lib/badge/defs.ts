import type { DetailedHTMLProps, HTMLAttributes, ReactNode } from 'react';
import type { BADGE_MOTIVES, BADGE_VARIANTS } from './consts';

type BadgeHTMLElementProps = Omit<
  DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
  'children'
>;

type BadgeMotive = (typeof BADGE_MOTIVES)[number];
type BadgeVariant = (typeof BADGE_VARIANTS)[number];

interface BadgeProps extends BadgeHTMLElementProps {
  children: ReactNode;
  motive?: BadgeMotive;
  variant?: BadgeVariant;
}

export type { BadgeProps, BadgeMotive, BadgeVariant };
