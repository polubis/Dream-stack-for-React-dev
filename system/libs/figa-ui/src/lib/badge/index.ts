export * from './badge';
export * from './defs';
