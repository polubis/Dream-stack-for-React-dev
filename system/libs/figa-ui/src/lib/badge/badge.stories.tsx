import type { Story, Meta } from '@storybook/react';

import { Badge } from './badge';
import { Box } from '../box';
import { BADGE_MOTIVES, BADGE_VARIANTS } from './consts';
import { Font } from '../font';

export default {
  component: Badge,
  title: 'Badge',
} as Meta;

const Template: Story = () => {
  return (
    <Box spacing={BADGE_VARIANTS.map(() => 300)} padding={[300, 300, 300, 300]}>
      {BADGE_VARIANTS.map((variant) => (
        <Box key={variant} spacing={BADGE_MOTIVES.map(() => 300)}>
          <Font variant="h5"> Variant {variant}</Font>
          {BADGE_MOTIVES.map((motive) => (
            <Badge motive={motive} variant={variant} key={motive}>
              BADGE {motive.toUpperCase()}
            </Badge>
          ))}
        </Box>
      ))}
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {};
