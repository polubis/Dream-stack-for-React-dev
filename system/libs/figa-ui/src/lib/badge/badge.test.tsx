import { fireEvent, render, screen } from '@testing-library/react';
import { Badge } from './badge';

describe('Badge can be used when: ', () => {
  it('[FRAGILE] works with default setup', () => {
    const { asFragment } = render(<Badge>My content inside</Badge>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = render(
      <Badge className="my-class" variant="outlined" motive="secondary">
        My content inside
      </Badge>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('allows to pass other native properties', () => {
    const spy = jest.fn();

    render(<Badge onClick={spy}>My content inside</Badge>);

    fireEvent.click(screen.getByText(/My content inside/));

    expect(spy).toHaveBeenCalledTimes(1);
  });
});
