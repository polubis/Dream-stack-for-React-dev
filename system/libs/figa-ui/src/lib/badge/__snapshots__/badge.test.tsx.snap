// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <div
    class="badge my-class outlined secondary"
  >
    My content inside
  </div>
</DocumentFragment>
`;

exports[`Badge can be used when:  [FRAGILE] works with default setup 1`] = `
<DocumentFragment>
  <div
    class="badge filled primary"
  >
    My content inside
  </div>
</DocumentFragment>
`;
